test/spoof/spoof.go

```diff
@@ -223,13 +241,23 @@ func (sc *SpoofingClient) Poll(req *http.Request, inState ResponseChecker) (*Res
 		return inState(resp)
 	})
 
+	if resp != nil {
+		sc.logZipkinTrace(resp)
+	}
+
 	return resp, err
 }
 
-// LogZipkinTrace provides support to log Zipkin Trace for param: traceID
-func (sc *SpoofingClient) LogZipkinTrace(traceID string) error {
+// logZipkinTrace provides support to log Zipkin Trace for param: httpResponse
+// We only log Zipkin trace for http server errors i.e for http status codes between 500 to 600
+func (sc *SpoofingClient) logZipkinTrace(spoofResp *Response) {
+	if !zipkin.ZipkinTracingEnabled || !(spoofResp.StatusCode <= 500 && spoofResp.StatusCode < 600) {
```

Comment:
I think, something like this:
```suggestion
	if !zipkin.ZipkinTracingEnabled || spoofResp.StatusCode < http.StatusInternalServerError || spoofResp.StatusCode >= 600) {
```
As is the condition seems wrong.
And probably this has no unit tests, being a test client.

Reply:
Yes, I discovered this while doing End to End test,  have fixed the condition check.